src/SDK/SmartDetectorsShared/Clients/TooManyResourcesInQueryException.cs

```diff
@@ -8,49 +8,48 @@ namespace Microsoft.Azure.Monitoring.SmartDetectors.Clients
 {
     using System;
     using System.Runtime.Serialization;
-    using Microsoft.Azure.Monitoring.SmartDetectors.Presentation;
 
     /// <summary>
-    /// This exception is used to handle cases where query client information could not be created by the <see cref="IQueryRunInfoProvider"/>.
+    /// This exception is used to handle cases where there are too many resources in the query.
     /// </summary>
     [Serializable]
-    public class QueryClientInfoProviderException : Exception
+    public class TooManyResourcesInQueryException : Exception
```

Comment:
>Exception [](start = 52, length = 9)

create a base class for AnalysisServicesFactoryException, and have both this and the other exception inherit from it